### PR TITLE
[DAT-1133] - Create a new page to update the payment information and reprocess the pending invoices

### PR DESCRIPTION
**Pull request by jhoffman-ms**
Create a new page to update the payment information and reprocess the pending invoices.

**Screenshots:**

**Update payment method:**
image

**Reprocess pending invoices:**
image

**Task:** [DAT-1133](https://makingsense.atlassian.net/jira/software/projects/DAT/boards/62?selectedIssue=DAT-1133)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ import DopplerIntlProvider from './i18n/DopplerIntlProvider';
 import { availableLanguageOrNull } from './i18n/utils';
 import { InjectAppServices } from './services/pure-di';
 import EditorsDemo from './components/EditorsDemo/EditorsDemo';
-
+import UpdatePaymentInformation from './components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/index';
```
Comment (andresmoschini) [inline src/App.js:34]:
```suggestion
import UpdatePaymentInformation from './components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/index';

```

File: src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  return (
+    <>
+      <table className="dp-c-table  m-t-24">
+        <thead>
+          <tr>
+            <th>
+              <a href="#">
+                {_('updatePaymentMethod.reprocess.invoices_declined_table.date_column')}
+              </a>
```
Comment (andresmoschini) [inline src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:32]: We are trying to use buttons styled like links in place of `a` when they are not real links.

File: src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            {_('updatePaymentMethod.reprocess.pending_amount_message')} {': '} {dollarSymbol}{' '}
+            <FormattedNumber value={state.declinedInvoices.totalPending} {...numberFormatOptions} />
+          </div>
+          <DeclinedInvoices declinedInvoices={state.declinedInvoices.invoices} />
+          {showMessage && (
+            <StatusMessage
+              type={status === SAVED ? 'success' : 'cancel'}
+              message={_(
+                status === SAVED ? 'updatePaymentMethod.reprocess.success_message' : messageError,
+              )}
```
Comment (andresmoschini) [inline src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:117]: We have not a translation for the returned `messageError`

File: src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      });
+    };
+    fetchData();
+  }, [dopplerBillingUserApiClient]);
+
+  const reprocess = async () => {
+    const result = await dopplerBillingUserApiClient.reprocess();
+
+    if (result.success) {
+      setStatus(SAVED);
```
Comment (andresmoschini) [inline src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:88]: Should you change the status or the _loading_ during the request?

File: src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js (hunk 3 above)
Comment (jhoffman-ms) [reply src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:117]: Yes, we have the translation inside ```first_data_error```

Comment (andresmoschini) [reply src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:117]: oh, grate!

in that case, what do you think about renaming  `messageError` as `errorMessageKey`?

File: src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js (hunk 2 above)
Comment (jhoffman-ms) [reply src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:32]: I will remove the ```a``` element. Becuase only we need to show the title in the header.

**Comment by ms-fcabrera** [comment]
👍

File: src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js (hunk 3 above)
Comment (jhoffman-ms) [reply src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:105]: Yes, I agree with you. I will change by errorMessageId

File: src/components/Plans/UpdatePaymentInformation/Reducers/reprocessReducer.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+export const INITIAL_STATE_REPROCESS = {
+  declinedInvoices: { invoices: [] },
```
Comment (jcamposmk) [inline src/components/Plans/UpdatePaymentInformation/Reducers/reprocessReducer.js:2]: Por qué no colocar `invoices` directamente? y que sea un array. O aparte de la propiedad invoices, también se usa algo más?

Comment (jcamposmk) [reply src/components/Plans/UpdatePaymentInformation/Reducers/reprocessReducer.js:2]: Si, ya vi que también se usa `totalPending`.

File: src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js [hunk 6]
```diff
@@ -10,6 +10,7 @@ import {
   reprocessReducer,
   REPROCESS_ACTIONS,
 } from '../../Reducers/reprocessReducer';
+import { UnexpectedError } from '../../../PlanCalculator/UnexpectedError/index';
```
Comment (jcamposmk) [inline src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:13]: hmmmm creo que hay un componente de `UnexpectedError` en la carpeta `shared` ... eventualmente, el de PlanCalculator habria que cambiarlo para que use el de la carpeta `shared`

Comment (jhoffman-ms) [reply src/components/Plans/UpdatePaymentInformation/UpdatePaymentInformation/Reprocess/index.js:13]: ahh, joya, ahi lo cambio. Gracias

**Comment by jcamposmk** [comment]
👍

**Pull request merged**